charts/argo-workflows/templates/controller/workflow-sa.yaml

```diff
@@ -1,13 +1,31 @@
 {{- if .Values.workflow.serviceAccount.create -}}
+
+  {{- /*
+      Cardinality-check the namespaces/namespace variables
+      This type check is done only once, here. It would be overkill to perform it for other RBAC resources as well.
+  */}}
+  {{- if and .Values.controller.workflowNamespaces .Values.workflow.namespace }}
+    {{- fail "Please set either controller.workflowNamespaces or workflow.namespace for the service account, not both" }}
+  {{- end }}
+
+  {{- /*
+      Type-check the namespaces variable for being a list ("slice")
+      This type check is done only once, here. It would be overkill to perform it for other RBAC resources as well.
+    */}}
+  {{- if .Values.controller.workflowNamespaces }}
+    {{- if not ( kindIs "slice" .Values.controller.workflowNamespaces ) }}
+      {{- fail ( printf "workflow.namespaces is a slice, got %s %s" ( kindOf .Values.controller.workflowNamespaces ) .Values.controller.workflowNamespaces ) }}
+    {{- end }}
+  {{- end }}
+
+  {{ range $namespace := append .Values.controller.workflowNamespaces .Values.workflow.namespace | compact | uniq }}
+---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ .Values.workflow.serviceAccount.name }}
-  {{- with .Values.workflow.namespace }}
-  namespace: {{ . }}
-  {{- end }}
-  {{- with .Values.workflow.serviceAccount.annotations }}
+  name: {{ $.Values.workflow.serviceAccount.name }}
+  namespace: {{ $namespace }}
   annotations:
-    {{- toYaml . | nindent 4 }}
+    {{- toYaml $.Values.workflow.serviceAccount.annotations | nindent 4 }}
```

Comment:
Please wrap this into `{{ with }}`, like before.

Reply:
Resolved in https://github.com/argoproj/argo-helm/pull/824/commits/232e72efbce5917c42db094a178cae15cb5151c5